<?xml version="1.0"?>
<robot name="robot_control" xmlns:xacro="http://www.ros.org/wiki/xacro">

<xacro:macro name="daleta_robot_control">

    <gazebo>
        <plugin filename="ign_ros2_control-system" name="ign_ros2_control::IgnitionROS2ControlPlugin">
        <robot_param>robot_description</robot_param>
        <robot_param_node>robot_state_publisher</robot_param_node>
        <!-- <parameters>$(find ign_ros2_control_demos)/config/cart_controller.yaml</parameters> -->
        </plugin>
    </gazebo>

    <!-- ros2_control -->
    <ros2_control name="IgnitionSystem" type="system">
        <hardware>
        <plugin>ign_ros2_control/IgnitionSystem</plugin>
        </hardware>
        <joint name="leg_1_joint_1">
            <command_interface name="position">
                <param name="min">-1</param>
                <param name="max">1</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>
        <joint name="leg_2_joint_1">
            <command_interface name="position">
                <param name="min">-1</param>
                <param name="max">1</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>
        <joint name="leg_3_joint_1">
            <command_interface name="position">
                <param name="min">-1</param>
                <param name="max">1</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>        
    </ros2_control>

  </xacro:macro>

</robot>